File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem67_CountingBits.java
package org.phoenix.leetcode.challenges;

public class Problem67_CountingBits {
    public int[] countBits(int num) {
        int[] arr = new int[num + 1];
        for (int i = num; num > 0; i--) {
            arr[i] = bits(num);
            num--;
        }
        return arr;
    }

    public int bits(int n) {
        if (n == 0) return 0;

        return (n & 1) + bits(n >> 1);
    }
}

File: JavaScript/leetcode75/src/main/Problem67_CountingBits.js
/**
 * @param {number} n
 * @return {number[]}
 */
var countBits = function (num) {
  let arr = new Array(num + 1).fill(0);
  for (let i = num; num > 0; i--) {
    arr[i] = bits(num);
    num--;
  }
  return arr;
};

let bits = function (n) {
  if (n == 0) return 0;

  return (n & 1) + bits(n >> 1);
};

module.exports = countBits;
